earth_enterprise/src/common/geFilePool.cpp

```diff
@@ -20,17 +20,65 @@
 #include <third_party/rsa_md5/crc32.h>
 #include <khEndian.h>
 
+// ****************************************************************************
+// ***  POSIXFileAcessor
+// ****************************************************************************
+int POSIXFileAccessor::Open(const std::string &fname, int flags, mode_t createMask) {
+  return khOpen(fname, flags, createMask);
+}
+
+int POSIXFileAccessor::FsyncAndClose(std::shared_ptr<AbstractFileIdentifier> pID) {
+  return khFsyncAndClose(pID->getAsFD());
+}
+
+int POSIXFileAccessor::Close(std::shared_ptr<AbstractFileIdentifier> pID) {
+  return khClose(pID->getAsFD());
+}
+
+bool POSIXFileAccessor::PreadAll(std::shared_ptr<AbstractFileIdentifier> pID, void* buffer, size_t size, off64_t offset) {
+  return khPreadAll(pID->getAsFD(), buffer, size, offset);
+}
+
+bool POSIXFileAccessor::PwriteAll(std::shared_ptr<AbstractFileIdentifier> pID, const void* buffer, size_t size, off64_t offset) {
+  return khPwriteAll(pID->getAsFD(), buffer, size, offset);
+}
+
+// ****************************************************************************
+// ***  FileIdentifierFactory
+// ****************************************************************************
+namespace FileIdentifierFactory {
+  static std::shared_ptr<AbstractFileIdentifier> getIdentifier(int i) {
+    return std::make_shared<POSIXFileIdentifier>(i);
```

Comment:
Minor: it might be more efficient to simply return the POSIXFileIdentifier rather than a pointer to it, i.e.,
```
return POSIXFileIdentifier(i);
```
That trades a copy for a bunch of pointer accesses, which I suspect is more efficient. The compiler may even be able to optimize out the copy.